test/e2e/network/ingressclass.go

```diff
@@ -50,13 +50,35 @@ var _ = common.SIGDescribe("IngressClass [Feature:Ingress]", func() {
 		framework.ExpectNoError(err)
 		defer deleteIngressClass(cs, ingressClass1.Name)
 
-		ingress, err := createBasicIngress(cs, f.Namespace.Name)
-		framework.ExpectNoError(err)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		lastFailure := ""
+
+		// the admission controller may take a few seconds to observe the ingress classes
+		if err := wait.Poll(time.Second, time.Minute, func() (bool, error) {
+			lastFailure = ""
+
+			ingress, err := createBasicIngress(cs, f.Namespace.Name)
+			if err != nil {
+				lastFailure = err.Error()
+				return false, err
+			}
+			defer func() {
+				err := cs.NetworkingV1().Ingresses(ingress.Namespace).Delete(ctx, ingress.Name, metav1.DeleteOptions{})
+				framework.Logf("%v", err)
+			}()
```

Comment:
this is cleaned by the garbage collector, isn' it? The e2e framewor creates and deletes the namespace after the execution of the test

Reply:
yeah, my bad, this run inside the poll function, I didn't realize